test_programs/compile_success_with_bug/underconstrained_value_detector_5425/src/main.nr

```diff
@@ -0,0 +1,37 @@
+unconstrained fn maximum_price(options: [u32; 3]) -> u32 {
+    let mut maximum_option = 0;
+    for option in options {
+        if (option > maximum_option) {
+            maximum_option = option;
+        }
+    }
+    maximum_option
+}
+
+fn main(sandwiches: pub [u32; 3], drinks: pub [u32; 3], snacks: pub [u32; 3], best_value: u32) {
+    let meal_deal_cost: u32 = 390;
+    let most_expensive_sandwich = maximum_price(sandwiches);
+    let mut sandwich_exists = false;
+    for sandwich_price in sandwiches {
+        assert(sandwich_price <= most_expensive_sandwich);
+        sandwich_exists |= sandwich_price == most_expensive_sandwich;
+    }
+    assert(sandwich_exists);
+
+    let most_expensive_drink = maximum_price(drinks);
+    let mut drink_exists = false;
+    for drink_price in drinks {
+        assert(drink_price <= most_expensive_drink);
+        drink_exists |= drink_price == most_expensive_drink;
+    }
+    assert(drink_exists);
```

Comment:
What is the difference between the sandwich and the drinks case? 👀 
I see that snacks don't have it, but do we need all three?

It would be great to have some comments in the code describing what the bug is, to make it easier for anyone not familiar with the rules to see what's going on 🙏

Reply:
Commented